analyses/mutational-signatures/mutational_signatures.Rmd

```diff
@@ -368,6 +369,247 @@ lapply(unique(nature_sigs_primary$short_histology),
+This is happening due to overfitting problems when fitting existing signatures.  De novo signature calling is used to provide an alternative for discussion. It is determined, which signatures can be extracted from the data themselves. These signatures are then fitted back to the known signatures that are used for the fitting approach above. 
+
+### De novo calling
+
+To prevent the loading of big data files into memory and intense processing, I have muted this part of the script and it loads the preprocessed parameters. 
+```{r}
+# determine the ideal number of signatures. 
+# the choice of 10 signatures is now hard coded, because otherwise it has to be rerun.
+# this step requires extensive ressources, the signature file is > 8 GB in size. Therefore a loop is included to load a preprocessed file, if available. 
+
+if(!file.exists("results/sigfit_signatures_10.rdat")){
+  
+  samples_extr <- extract_signatures(counts = sigs_input,
+                                        nsignatures = 5:15,
+                                        iter = 1000, 
+                                        seed = 42)
+  dev.off()
+  
+  samples_extr10 <- extract_signatures(counts = sigs_input,
+                                        nsignatures = 10,
+                                        iter = 10000, 
+                                        seed = 42)
+    
+  save(samples_extr10, file="results/sigfit_signatures_10.rdat")
+}
+```
+
+```{r}
+if(!file.exists("results/sigfit_signatures_10_signatures.rdat")){
+  load("results/sigfit_signatures_10.rdat")
+  signatures <- retrieve_pars(samples_extr10,
+                            par = "signatures")
+  save(signatures,file="results/sigfit_signatures_10_signatures.rdat")
+}else{
+   load("results/sigfit_signatures_10_signatures.rdat")
+}
+
+if(!file.exists("results/sigfit_signatures_10_exposures.rdat")){
+  load("results/sigfit_signatures_10.rdat")
+  exposures <- retrieve_pars(samples_extr10,
+                            par = "exposures")
+  save(exposures,file="results/sigfit_signatures_10_exposures.rdat")
+}else{
+   load("results/sigfit_signatures_10_exposures.rdat")
+}
+
+if(!file.exists("results/sigfit_signatures_10_reconstructions.rdat")){
+  load("results/sigfit_signatures_10.rdat")
+  reconstructions <- retrieve_pars(samples_extr10,
+                            par = "reconstructions")
+  save(reconstructions,file="results/sigfit_signatures_10_reconstructions.rdat")
+}else{
+   load("results/sigfit_signatures_10_reconstructions.rdat")
+}
+  
+  recmean <- reconstructions$mean
+  rownames(recmean)<- rownames(sigs_input) 
+  expmean <- exposures$mean
+  explow95 <- exposures$lower_95
+  rownames(expmean)<- rownames(sigs_input)
+  
+  # set exposures to 0, if lower confidence interval hits 1%
+  expmean[explow95<0.01]<-0
+```
```

Comment:
If I'm understanding correctly - if we're in a situation where the underlying data files change (or the software dependencies!), I have the large files locally, and I know that everything related to mutational signatures needs to be rerun, I'd have to manually delete the files to accomplish that. I agree that skipping long-running, memory intensive steps by default makes a lot of sense, but I am worried about the situation I described.

To accommodate both situations, I propose we control the behavior here using [`params`](https://rmarkdown.rstudio.com/lesson-6.html), where there is some `params` that specifies whether or not to check for existing files and it checks for existing files by default.

I think before making any kind of specific recommendation, I'd like to understand this a bit more - is there ever a scenario where we would expect _some but not all_ of the files to exist?

Reply:
I agree that a solution with params would be best. My solution is not ideal. 
If there are new data, everything would need to be rerun, because the signatures are derived from the cohort. This is the big disadvantage over using a fitting approach.